src/sw.ts

```diff
@@ -0,0 +1,130 @@
+import { MESSAGE_TYPES as MESSAGE_TYPES_, Message, FileRecord } from './lib/types';
+import * as CL from 'comlink';
+
+importScripts('./lib/comlink.js');
+
+const swScope = self as ServiceWorkerGlobalScope;
+const Comlink = (swScope as any).Comlink as typeof CL;
+
+const MESSAGE_TYPES = {
+  ESTABLISH_HANDSHAKE: "ESTABLISH_HANDSHAKE",
+  HANDSHAKE_RECEIVED: "HANDSHAKE_RECEIVED",
+} as unknown as typeof MESSAGE_TYPES_;
```

Comment:
It feels unfair that you should have to type these out again like this.

I don't know off-hand if there's a better solution, but I thought you should know I think that's unfair.

Reply:
I had a similar issue in https://github.com/Polymer/koa-node-resolve/blob/master/src/support/logger.ts with LogLevel.

Reply:
It would be great if sw modules were here. Perhaps I should compile the SW into a single file?